frontend/src/metabase/parameters/utils/mapping-options.js

```diff
@@ -1,30 +1,38 @@
 import { tag_names } from "cljs/metabase.shared.parameters.parameters";
 import { isActionDashCard } from "metabase/actions/utils";
 import { isVirtualDashCard } from "metabase/dashboard/utils";
+import { getColumnIcon } from "metabase/common/utils/columns";
+import { getColumnGroupName } from "metabase/common/utils/column-groups";
+import * as Lib from "metabase-lib";
 import Question from "metabase-lib/Question";
-import { ExpressionDimension } from "metabase-lib/Dimension";
 import {
+  columnFilterForParameter,
   dimensionFilterForParameter,
   getTagOperatorFilterForParameter,
   variableFilterForParameter,
 } from "metabase-lib/parameters/utils/filters";
 import {
+  buildColumnTarget,
   buildDimensionTarget,
   buildTemplateTagVariableTarget,
   buildTextTagTarget,
 } from "metabase-lib/parameters/utils/targets";
 
-function buildStructuredQuerySectionOptions(section) {
-  return section.items.map(({ dimension }) => ({
-    sectionName: section.name,
-    name: dimension.displayName(),
-    icon: dimension.icon(),
-    target: buildDimensionTarget(dimension),
-    // these methods don't exist on instances of ExpressionDimension
-    isForeign: !!(dimension instanceof ExpressionDimension
-      ? false
-      : dimension.fk() || dimension.joinAlias()),
-  }));
+function buildStructuredQuerySectionOptions(query, stageIndex, group) {
+  const groupInfo = Lib.displayInfo(query, stageIndex, group);
+  const columns = Lib.getColumnsFromColumnGroup(group);
+
+  return columns.map(column => {
+    const columnInfo = Lib.displayInfo(query, stageIndex, column);
+
+    return {
+      sectionName: getColumnGroupName(groupInfo),
+      name: columnInfo.displayName,
+      icon: getColumnIcon(column),
+      target: buildColumnTarget(column),
+      isForeign: columnInfo.isFromJoin || columnInfo.isImplicitlyJoinable,
```

Comment:
It's a good opportunity to add JSDoc to `isImplicitlyJoinable` saying that it's `true` when column has a FK.